src/compile/model.ts

```diff
@@ -233,7 +233,11 @@ export abstract class Model {
   }
 
   public renameData(oldName: string, newName: string) {
-     this._dataNameMap.rename(oldName, newName);
+    if (oldName === newName) {
```

Comment:
Also, you already have `console.error` method inside `rename` method of the name map.  
Not sure why you have to add it twice here.